### DISCUSSION TITLE
treesitter parity with org elements?

### DISCUSSION THREAD

**Discussion by sotte** [nvim-orgmode/orgmode#579] [General]
I was looking into treesitter and how the different org elements are represented. I was hoping that there was an easy way to get all the e.g. links.

Here is an example of `:InspectTree` for a simple org file:
![image](https://github.com/nvim-orgmode/orgmode/assets/79138/cf67b09a-c19f-432e-906c-e86e97218c91)

I'm mostly wondering if there is the goal to have parity between treesitter and org elements? This would allow the very generic treesitter queries to retrieve all the elements of orgmode.

**Comment by jgollenz** [comment]
The treesitter grammar for org-mode lives [here](https://github.com/milisims/tree-sitter-org). The maintainer made [a case against it](https://github.com/milisims/tree-sitter-org/issues/25#issuecomment-1139741645). 
I think we could still achieve what you describe with their suggestion: 

> write an object-parser as a tree-sitter grammar, and inject it in the appropriate locations

If you are knowledgeable in this domain you are very welcome to share your ideas and open up a PR :+1:

**Reply by sotte** [reply]
Thanks @jgollenz.
I'm just playing around with treesitter and I'm not familiar with "object parsers". After reading the docs  (`:help treesitter`) I still don't have a clue what this is. They talk about "parser object" twice, but I'm not sure if this is even the same.

I just ended up writing a simple regex to get the job done.

It would be quite cool though to have this functionality. "Give me all all headlines which contain links" or similar.
